src/lib.rs

```diff
@@ -283,124 +179,84 @@ impl Renderer {
             MinDepth: 0.0,
             MaxDepth: 1.0,
         };
-        ctx.RSSetViewports(1, &vp);
-
+        let draw_fmt = if mem::size_of::<DrawIdx>() == 2 { DXGI_FORMAT_R16_UINT } else { DXGI_FORMAT_R32_UINT };
         let stride = mem::size_of::<DrawVert>() as u32;
-        ctx.IASetInputLayout(self.input_layout.as_raw());
-        ctx.IASetVertexBuffers(0, 1, &self.vertex_buffer.as_raw(), &stride, &0);
-        ctx.IASetIndexBuffer(
-            self.index_buffer.as_raw(),
-            if mem::size_of::<DrawIdx>() == 2 {
-                DXGI_FORMAT_R16_UINT
-            } else {
-                DXGI_FORMAT_R32_UINT
-            },
-            0,
-        );
+        let blend_factor = 0.0;
+
+        ctx.RSSetViewports(&[vp]);
+        ctx.IASetInputLayout(&self.input_layout);
+        ctx.IASetVertexBuffers(0, 1, &self.vertex_buffer.0, &stride, &0);
+        ctx.IASetIndexBuffer(self.index_buffer.get_buf(), draw_fmt, 0);
         ctx.IASetPrimitiveTopology(D3D11_PRIMITIVE_TOPOLOGY_TRIANGLELIST);
-        ctx.VSSetShader(self.vertex_shader.as_raw(), ptr::null(), 0);
-        ctx.VSSetConstantBuffers(0, 1, &self.constant_buffer.as_raw());
-        ctx.PSSetShader(self.pixel_shader.as_raw(), ptr::null(), 0);
-        ctx.PSSetSamplers(0, 1, &self.font_sampler.as_raw());
-        ctx.GSSetShader(ptr::null_mut(), ptr::null(), 0);
-        ctx.HSSetShader(ptr::null_mut(), ptr::null(), 0);
-        ctx.DSSetShader(ptr::null_mut(), ptr::null(), 0);
-        ctx.CSSetShader(ptr::null_mut(), ptr::null(), 0);
-
-        let blend_factor = [0.0; 4];
-        ctx.OMSetBlendState(self.blend_state.as_raw(), &blend_factor, 0xFFFFFFFF);
-        ctx.OMSetDepthStencilState(self.depth_stencil_state.as_raw(), 0);
-        ctx.RSSetState(self.rasterizer_state.as_raw());
+        ctx.VSSetShader(&self.vertex_shader, &[]);
+        ctx.VSSetConstantBuffers(0, &[Some(self.constant_buffer.clone())]);
+        ctx.PSSetShader(&self.pixel_shader, &[]);
+        ctx.PSSetSamplers(0, &[Some(self.font_sampler.clone())]);
+        ctx.GSSetShader(None, &[]);
+        ctx.HSSetShader(None, &[]);
+        ctx.DSSetShader(None, &[]);
+        ctx.CSSetShader(None, &[]);
+        ctx.OMSetBlendState(&self.blend_state, &blend_factor, 0xFFFFFFFF);
+        ctx.OMSetDepthStencilState(&self.depth_stencil_state, 0);
+        ctx.RSSetState(&self.rasterizer_state);
     }
 
-    unsafe fn create_vertex_buffer(
-        device: &ComPtr<ID3D11Device>,
-        vtx_count: usize,
-    ) -> HResult<Buffer> {
+    unsafe fn create_vertex_buffer(device: &ID3D11Device, vtx_count: usize) -> Result<Buffer> {
         let len = vtx_count + VERTEX_BUF_ADD_CAPACITY;
         let desc = D3D11_BUFFER_DESC {
             ByteWidth: (len * mem::size_of::<DrawVert>()) as u32,
             Usage: D3D11_USAGE_DYNAMIC,
-            BindFlags: D3D11_BIND_VERTEX_BUFFER,
-            CPUAccessFlags: D3D11_CPU_ACCESS_WRITE,
+            BindFlags: 1,
+            CPUAccessFlags: 65536,
```

Comment:
I assume these aren't exposed by `windows`? Can you make these named constants in that case, otherw no one knows what they stand for

Reply:
Associated bug here if you care: https://github.com/microsoft/windows-rs/issues/1741